### PR TITLE
[PAY-1363] DMs: Add CTA to Create Chat user rows

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds a call to action to users that require some action to unlock tipping, eg if they're tip gated or if you've blocked them. Clicking a "disabled" user brings up the "InboxUnavailable" modal. Closing the "InboxUnavailable" modal should take you back to the search without losing its state.
image

https://github.com/AudiusProject/audius-client/assets/3690498/b3863312-dc57-4adb-b5b7-f07a71493e98

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

Opening the tip modal loses context and you can't get back to search, nor can you start the chat with the user

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx [hunk 1]
```diff
@@ -122,7 +129,7 @@ export const InboxUnavailableModal = ({
     <Modal
       bodyClassName={styles.modalBody}
       isOpen={isVisible}
-      onClose={onClose}
+      onClose={handleDismiss}
```
Comment (rickyrombo) [inline packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx:132]: differentiate between closing the inbox unavailable modal and "dismissing" - closing without action

File: packages/web/src/components/search-users-modal/SearchUsersModal.tsx [hunk 2]
```diff
@@ -106,6 +99,7 @@ export const SearchUsersModal = (props: SearchUsersModalProps) => {
 
   const handleClose = useCallback(() => {
     setIsVisible(false)
+    setQuery('')
```
Comment (rickyrombo) [inline packages/web/src/components/search-users-modal/SearchUsersModal.tsx:102]: only clear the query if dismissed, not if hiding temporarily for nested modals

File: packages/web/src/pages/chat-page/components/CreateChatModal.tsx [hunk 3]
```diff
@@ -54,23 +76,34 @@ export const CreateChatModal = () => {
...
+          <MessageUserSearchResult
+            key={user.user_id}
+            user={user}
+            openInboxUnavailableModal={handleOpenInboxUnavailableModal}
+            closeParentModal={closeParentModal}
+          />
+        )}
+        renderEmpty={() => <CreateChatEmptyResults />}
+      />
+      {user ? (
```
Comment (rickyrombo) [inline packages/web/src/pages/chat-page/components/CreateChatModal.tsx:99]: this part is the new part, the rest is just wrapping in a fragment

File: packages/web/src/pages/modals/Modals.tsx [hunk 4]
```diff
@@ -91,6 +90,7 @@ const Modals = () => {
       <UnloadDialog />
       <CollectibleDetailsModal />
       <StripeOnRampModal />
+      <CreateChatModal />
```
Comment (rickyrombo) [inline packages/web/src/pages/modals/Modals.tsx:93]: Don't unmount the modal when hiding it so it retains state

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-create-chat-modal-cta

File: packages/web/src/pages/modals/Modals.tsx (hunk 4 above)
Comment (dharit-tan) [reply packages/web/src/pages/modals/Modals.tsx:93]: Will this save state if user dismisses and then later comes back? ie. are you clearing state on dismiss?

Comment (dharit-tan) [reply packages/web/src/pages/modals/Modals.tsx:93]: I see clearing the query, close vs dismiss getting confusing

**Review by dharit-tan — APPROVED** [review]

**Pull request merged**